### PR TITLE
Adding tools for development

### PR DESCRIPTION
**Pull request by maelk**
Adding:
- repo management scripts : init and update
- test script for baremetal provider
- Dockerfile and script to run a development container

### CODE REVIEW

File: container/run-workspace.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -ue
+
+SCRIPTPATH=$( cd $(dirname $0) >/dev/null 2>&1 ; pwd -P )
```
Comment (kanwar-saad) [inline container/run-workspace.sh]: This causes problems of relative paths. you need this to get the absolute path.

"$(dirname "$(readlink -f "$0")")"

File: tools/init-repo.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -ue
+
+SCRIPTPATH=$( cd $(dirname $0) >/dev/null 2>&1 ; pwd -P )
```
Comment (kanwar-saad) [inline tools/init-repo.sh]: same here

File: tools/init-repo.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  git remote -v
+  # Update "master" on Nordix
+  git fetch upstream
+  git rebase upstream/master
+  git push
+  cd ..
+}
+
+setup_repo cluster-api kubernetes-sigs
+for repo in metal3-dev-env cluster-api-provider-baremetal baremetal-operator
```
Comment (kanwar-saad) [inline tools/init-repo.sh]: this is fine for now .. but this script can be modified slightly to make it generic and work for other repos as well.

File: container/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    mv ./kubectl /usr/local/bin/kubectl
+
+ENV GOPATH=/root/go
+ENV PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/root/go/bin
+
+RUN curl -LO https://raw.githubusercontent.com/Nordix/cluster-api-provider-baremetal/master/tools/install_kustomize.sh && \
+    chmod +x ./install_kustomize.sh && ./install_kustomize.sh && rm install_kustomize.sh && \
+    curl -LO https://raw.githubusercontent.com/Nordix/cluster-api-provider-baremetal/master/tools/install_kubebuilder.sh && \
+    chmod +x ./install_kubebuilder.sh && ./install_kubebuilder.sh && rm install_kubebuilder.sh && \
+    mv kubebuilder /usr/local
```
Comment (namnx228) [inline container/Dockerfile]: The `install_kustomize.sh` only installs Kustomize version `1.0.11` but the latest is `3.0.3`. Should we use the newer ?

File: container/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,40 @@
+FROM ubuntu:latest
```
Comment (kanwar-saad) [inline container/Dockerfile]: this usually breaks in CI when there is a new version of ubuntu released with some different package name. locking it to a particular version would be better. like 18.04 tag is updated frequently with security updates and bug fixes so maybe thats fine.

File: container/Dockerfile (hunk 4 above)
Comment (maelk) [reply container/Dockerfile]: maybe we could do it in the baremetal-operator kustomize install script. I would not change it here

**Pull request merged**